src/marvin/utilities/openai.py

```diff
@@ -22,33 +28,90 @@ def get_openai_client() -> AsyncClient:
         client = get_client()
         ```
     """
-    from marvin import settings
 
-    api_key: Optional[str] = (
-        settings.openai.api_key.get_secret_value() if settings.openai.api_key else None
-    )
-    organization: Optional[str] = settings.openai.organization
+    kwargs = {}
+
+    # --- Openai
+    if marvin.settings.provider == "openai":
+        client_class = AsyncClient if is_async else Client
+
+        api_key = marvin.settings.openai.api_key
+        if isinstance(api_key, pydantic.SecretStr):
+            api_key = api_key.get_secret_value()
```

Comment:
nit: is this check needed? im pretty sure no matter how you set it, [it should be a `SecretStr`](https://github.com/PrefectHQ/marvin/blob/main/src/marvin/settings.py#L20-L32) by the time we look here

Reply:
Empirically yes it is but now that you mention it I don't know why... but I can get this to crash without that. Let me look into it, something must be weird

Reply:
https://github.com/PrefectHQ/marvin/pull/772/commits/0d993d6f178af8ee1b1dd57a96cb13cea2a44a4b